### DISCUSSION TITLE
Compile tailwind css stylesheet as a string from a given string containing html with tailwind classes

### DISCUSSION THREAD

**Discussion by LiChoi** [rails/tailwindcss-rails#417] [Q&A]
Let's say I have a bunch of html files stored in the database.
Each one is using tailwind classes.
How do I compile the tailwind css stylesheet for each file and then store this in an associated table whenever the files change?

Use case: each file is a template, and templates can be put together to build different pages, so there is no way to build 1 tailwind css style sheet that works for the entire app. Each page needs a different tailwind css stylesheet, and this needs to be built off the stylesheet of the individual templates for that page.

**Comment by flavorjones** [comment]
This gem (and now the gem [`tailwindcss-ruby`](https://github.com/flavorjones/tailwindcss-ruby)) uses the command-line tool from the upstream project https://github.com/tailwindlabs/tailwindcss. That tool operates on files (both input and output) and I don't know of any options that would input on strings (like stdin/stdout).

Perhaps this question might be better asked upstream on the tailwindcss project?

**Reply by LiChoi** [reply]
Thank you for responding. I have posted the same question on tailwindcss project. There appears to be a way to do this in javascript. In tailwind documentation https://tailwindcss.com/docs/content-configuration it mentions we can pass raw html content in the configuration. Somebody had suggested the following might be possible:

```

const postcss = require('postcss');
const tailwindcss require('tailwindcss');
const baseConfig = require('./tailwind.config');

const sourceCSS = '@tailwind base; @tailwind components; @tailwind utilities';
const config = {
  presets: [baseConfig],
  content: [{ raw: markup }],
};
postcss([tailwindcss(config)])
  .process(sourceCSS)
  .then((css) => {
    // `css` is a string of the compiled CSS.
  });

```
